compiler/src/main/java/dagger/internal/codegen/BindingGraph.java

```diff
@@ -210,14 +215,26 @@ private BindingGraph create(
         explicitBindingsBuilder.add(provisionBindingFactory.forComponent(componentDependency));
         List<ExecutableElement> dependencyMethods =
             ElementFilter.methodsIn(elements.getAllMembers(componentDependency));
+        HashMultimap<Key, Equivalence.Wrapper<TypeMirror>> componentMethodType = HashMultimap.create();
```

Comment:
Why do we care what the name of the method is? The name of the method has nothing to do with its key. I would think we want to handle the case where:
* same return type and qualifier
* no args
* no parameters